pkg/controller/controllername/controllername.go

```diff
@@ -0,0 +1,48 @@
+package controllername
+
+import (
+	"strings"
+	"unicode"
+)
+
+const (
+	metricsNameDelimiter = "_"
+	loggerNameDelimiter  = "-"
+)
+
+type ControllerNamer interface {
+	MetricsName() string
+	LoggerName() string
+}
+
+// controllerName ex. {"My","Controller", "Name"} ->  MyControllerName
+type controllerName []string
+
+func NewControllerName(name []string) controllerName {
+	cn := make(controllerName, len(name))
+
+	for i, w := range name {
+		cn[i] = Strip(strings.ToLower(w))
+
+	}
+	return cn
+}
+
+// Strip removes spaces and non letters
+func Strip(s string) string {
+	rr := make([]rune, 0, len(s))
+	for _, r := range s {
+		if !unicode.IsSpace(r) && unicode.IsLetter(r) {
```

Comment:
do we need the isspace check when we have isletter?

Reply:
no i removed it